### ISSUE TITLE
[Cleanup]: Fix Android Studio Warnings

### ISSUE THREAD

**Issue by david-allison** [ankidroid/Anki-Android#13282]
This task is designed as a small, useful task to onboard new users to the codebase who are unsure about what  to work on.

> [!IMPORTANT] 
> **Unique constraint to this issue**
> Multiple pull requests from the same author will not be accepted, unless the refactoring is part of a larger piece of work 

> [!TIP]
> You do not need to be assigned this issue to work on it

Android Studio flags a number of warnings in the AnkiDroid project.

If a file which contains these warnings is committed, then Android Studio will warn the author. If the warning was not introduced by the author's commit, then the author is unnecessarily slowed down by having to confirm that they did not introduce more warnings into the codebase (which inevitably leads them to disabling the warning, as they're low value).

The list of warnings can be obtained via:
* Code -> Inspect Code

![Screenshot 2023-02-16 at 21 38 16](https://user-images.githubusercontent.com/62114487/219492902-6f1198ee-50ed-4cf5-9d7b-95816653d8e8.png)

The aim of this task is to introduce new contributors to the code with an 'easy' first issue to resolve.

1. Run `Code -> Inspect Code` on the solution (it will be slow)
2. Select one file which has warnings (under the `Kotlin` header is easiest.- many of the Java warnings are not flagged by the IDE)
3. Fix all the errors, warnings + weak warnings in the file, automatically if this is offered by the IDE, or manually otherwise.
4. Produce a Pull Request

If you have completed this and desire to do more work, please select one of our accepted issues, [or fix some `@NeedsTest` annotations in the codebase.](https://github.com/ankidroid/Anki-Android/issues/13283)

Once this issue has been completed, we can use CI to run Android Studio in headless mode to ensure that the codebase gets to, and stays at 0 warnings.

**Comment by atakanserifoglu** [comment]
You can assign me to this issue please

**Comment by lukstbit** [comment]
@atakanserifoglu Please read carefully the constraints mentioned above:

_Multiple pull requests from the same author for this issue will not be accepted (this is a constraint unique to this issue)._

**Comment by atakanserifoglu** [comment]
The project's naming convention has a lot of underscores whereas all of the issues related to this topic suggests camel case convention. Should I adapt all of these functions, classes and variable names or look for a different solution?

**Comment by david-allison** [comment]
> The project's naming convention has a lot of underscores whereas all of the issues related to this topic suggests camel case convention. Should I adapt all of these functions, classes and variable names or look for a different solution?

If they're in `libanki`, ignore naming conventions. Otherwise they're candidates to be removed

**Comment by Prachi-Jamdade** [comment]
Hey @david-allison, can I start working on this issue ?

**Comment by david-allison** [comment]
Feel free to!

**Comment by deekshatw** [comment]
@david-allison Can I work on this issue?

**Comment by lukstbit** [comment]
@deekshatw and others. Go ahead, there's no need to ask for permission to work on this issue. 
Follow the steps above, pick a file that you want to fix, check the [active PRs list](https://github.com/ankidroid/Anki-Android/pulls) to see if that specific file doesn't already have a PR associated, make your own PR, be happy.

**Comment by Prachi-Jamdade** [comment]
I was trying to fix some of the warnings. But then I'm facing the build issue while pushing my code. I haven't made any production code changes. @david-allison could you please guide me on this 

Here is the screenshot of what went wrong -

![image](https://user-images.githubusercontent.com/94190717/222894425-191d2681-9cba-4008-b21a-fbbf475fda67.png)

**Comment by david-allison** [comment]
> I was trying to fix some of the warnings. But then I'm facing the build issue while pushing my code. I haven't made any production code changes. @david-allison could you please guide me on this
> 
> Here is the screenshot of what went wrong -
> 
> ![image](https://user-images.githubusercontent.com/94190717/222894425-191d2681-9cba-4008-b21a-fbbf475fda67.png)

Add an issue, and come to [Discord](https://discord.gg/qjzcRTx)

Very likely related to: https://github.com/ankidroid/Anki-Android/pull/13280

**Comment by TusharBhatia34** [comment]
is this issue still open ? can i work on it?

**Comment by lukstbit** [comment]
> is this issue still open ?

Yes, it is. You can make a PR fixing one of the many reported issues.

**Comment by vreoo** [comment]
Hi @BrayanDSO ,

I noticed that my recent PR, which was merged a few days ago, automatically closed this issue as I mentioned it fixes this ticket. However, it seems this issue should remain open, as it has been tagged with "Keep Open." Could you kindly consider reopening it? Thank you!

**Comment by rubenquadros** [comment]
Hey, I will be cleaning up this part!
<img width="740" alt="Screenshot 2023-10-07 at 4 11 23 PM" src="https://github.com/ankidroid/Anki-Android/assets/31680582/0f261177-7f73-4a9a-bb03-8ac1a27675f5">

**Comment by neeldoshii** [comment]
> The list of warnings can be obtained via:
> 
> Analyze -> Inspect Code

<img width="154" alt="image" src="https://github.com/ankidroid/Anki-Android/assets/60827173/0ec034b2-8a14-47a7-8ac5-5303cdd980a5">

In newer version of Android Studio it has been changed from `Analyze ---> Inspect Code` to `Code ---> Inspect Code`. 

I think it's good to add in description of the issue for people who's using newer version of the Android Studio.

**Comment by SahityaGupta25** [comment]
Hi @david-allison ,
Thanks for suggesting this issue! 🙏
I’d like to work on this one as my first contribution.
Could you please assign it to me?

**Comment by david-allison** [comment]
> You do not need to be assigned this issue to work on it

**Comment by Kota-Jagadeesh** [comment]
Hey @david-allison I would like to take up the issue and fix warnings in one of the file as a  first contribution.
